## Abstract {.page_break_before}

Metagenomics has expanded our knowledge of microbial diversity, but contaminant sequences are frequently accidentally included in metagenome-assembled genomes. 
Genome contamination is often estimated by the presence of marker genes that are biased against detecting contaminants lacking these sequences. 
Further, most contamination detection tools do not remove contamination. 
We present charcoal, a tool that rapidly identifies and removes contamination in metagenome-assembled genomes using k-mer based methods.
K-mers are nucleotide sequences of length k. 
Sufficiently long k-mers are usually specific to a taxonomic lineage. 
Taking advantage of this property of k-mers, charcoal identifies majority and minority lineages for each contiguous sequence in a genome and removes contiguous sequences belonging to minority lineages when those lineages occur below a taxonomic threshold (by default, order). 
Applying charcoal to the Genome Taxonomy Database rs207, we found approximately XX% of genomes in GTDB were contaminated, with contamination broadly distributed across species and occurring in representative and RefSeq genomes. 
Genomes with longer contiguous sequences were less likely to be contaminated.
Our results show concordance with CheckM on detecting the presence of contamination in a genome. 
Charcoal is a snakemake workflow developed around the tool sourmash. 
It is available at github.com/dib-lab/charcoal, and is pip installable.