# Introduction

Metagenomic sequencing has expanded our knowledge of microbial communities and their diversity [@doi:10.1038/nmicrobiol.2016.48; @doi:10.1038/s41587-020-0718-6; @doi:10.1038/s41564-021-00928-6].
*De novo* metagenome analysis has generated thousands of draft genomes, termed metagenome-assembled genomes (MAGs), from organisms from diverse environments [@doi:10.1038/nature02340; @doi:10.1016/j.cell.2019.01.001; @doi:10.1038/s41587-020-0603-3; @doi:10.1038/s41587-020-0718-6].
Recently, large-scale re-analysis efforts have led to a rapid expansion in draft genomes in public repositories like the European Nucleotide Archive [@doi:10.1038/s41587-020-0603-3] and the Joint Genome Institute IMG/M [@doi:10.1038/s41587-020-0718-6].
Increased observation of draft genomes across the tree of life better enables researchers to contextualize new sequencing data and the roles that microorganisms play in diverse metabolic processes [@doi:10.1038/s41564-017-0012-7; @doi:10.1093/nar/gkz1002].

MAG inference relies on assembly and binning of metagenomic sequencing data.
Assembly produces long contiguous sequences by identifying overlaps between short sequencing reads, while binning groups assembled sequences into MAGs using read coverage and tetramernucleotide frequency.
Both processes are subject to biases that can reduce the completeness of or increase the contamination in a MAG: low sequencing coverage or high genomic variation causes short read assemblers to break contiguous sequences into shorter pieces (CITE), which decreases the signal for and accuracy of binning (CITE).
Commonly, the completeness and purity of MAGs is estimated through the presence and sequence composition of single-copy marker genes [@doi:10.1101/gr.186072.114] [@doi:10.1093/bioinformatics/btv351], with MAGs that reach >90% completeness and <5% contamination considered high quality [@doi:10.1101/gr.186072.114].
Single-copy marker genes are sets of genes that are present once in a genome of almost all members of a taxonomic group [@doi:10.1101/gr.186072.114; @doi:10.7717/peerj.243].
Using these genes to estimate contamination leads to two important biases.
First, given the assumption that marker genes are universally present in genomes, if a marker gene resides on a contaminant sequence but no other sequence for that marker gene is present in the genome, it will not be detected as contamination [@doi:10.1101/gr.186072.114; @doi:10.3389/fmicb.2017.02264].
When a MAG is substantially complete, this may lead to a small underestimation in contamination (~3% [@doi:10.1101/gr.186072.114]), but as completeness decreases, contamination may be substantially underestimated [@doi:10.3389/fmicb.2017.02264].
Second, contiguous sequences which do not contain marker sequences are not included among contamination estimates [@doi:10.1101/gr.186072.114] (note checkm called out plasmids and phages for this bias specifically).

Given these biases, methods that do not rely solely on marker genes may be better suited for contamination estimation.
Many tools have recently been developed that rely on different strategies for detecting contamination [@doi:10.1186/s13059-022-02619-9].
GUNC expands beyond marker genes and uses the full complement of genes in a genome to identify contiguous sequences that fall outside of the predicted lineage [@doi:10.1186/s13059-021-02393-0].
It estimates the distribution of taxonomic assignments within and across contiguous sequences to identify contaminants even among sequences that are poorly taxonomically labelled [@doi:10.1186/s13059-021-02393-0].
Conterminator detects cross-kingdom contamination using all-vs-all alignment and is geared toward contamination detection in large collections of genomes (e.g. databases) [@doi:10.1186/s13059-020-02023-1].
Sequences are considered contaminants when at least 100 nucleotides and no more than 20 kb share a sequence identity of at least 90% in genomes that originate from different kingdoms [@doi:10.1186/s13059-020-02023-1].
In addition, conterminator finds contamination in protein sequences by identifying protein clusters that contain sequences from cross-kingdom members [@doi:10.1186/s13059-020-02023-1].
 
Removing contamination is a separate problem from estimating its extent.
Tools like RefineM [@doi:10.1038/s41564-017-0012-7], MAGPpurify [@doi:10.1038/s41586-019-1058-x], and BlobTools [@doi:10.12688/f1000research.12232.1] rely on a combination of GC content, tetramernucleotide frequency, read coverage, phylogenetic or clade markers, conspecific sequences, and known contaminants to flag contaminant contiguous sequences for removal.
The inclusion of read coverage profiles necessitates the use of the original sequencing reads to produce coverage profiles in BAM format for the genomes.
Given that sequencing data and BAM files are usually orders of magnitude larger than genome sequences, this requirement limits the utility of such approaches, especially for large-scale genome databases. 
Tools like GUNC are poised for contamination removal without relying on read coverage profiles, but as of yet such approaches are unimplemented [@doi:10.1186/s13059-021-02393-0].

Long k-mers capture relatedness between organisms, where a k=31 captures species-level similarity [@doi:10.1128/mSystems.00020-16].
K-mers offer an alternative metric to identify contamination, especially in sequences lacking marker genes. 
Here we describe Charcoal, an automated method for filtering contaminant contiguous sequences from MAGs. 
We show...
We show...
We envisage that charcoal will complement marker gene-based approaches for contamination estimation, removing problematic sequences before they are further analyzed or propagated in public databases. 
